libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/UpcomingRoadObject.kt

```diff
@@ -1,25 +1,34 @@
 package com.mapbox.navigation.base.trip.model.roadobject
 
+import com.mapbox.navigation.base.trip.model.roadobject.distanceinfo.RoadObjectDistanceInfo
+
 /**
- * Holds the road objects and the distance to the point where the alert occurs,
- * or start an alert if it has length.
+ * Holds the road objects and the distance to the start of the object.
+ * There are two sources of road objects: active route and the electronic horizon.
+ * Road objects coming from the electronic horizon will also have [RoadObjectDistanceInfo]
+ * Available [RoadObjectDistanceInfo] types are:
+ * - [RoadObjectDistanceInfoType.GANTRY]
+ * - [RoadObjectDistanceInfoType.LINE]
+ * - [RoadObjectDistanceInfoType.POINT]
+ * - [RoadObjectDistanceInfoType.POLYGON]
+ * - [RoadObjectDistanceInfoType.SUB_GRAPH]
  *
  * @param roadObject road object
- * @param distanceToStart distance to the start of the alert.
- * If the object has a length, and we've passed the start point,
- * **this value will be negative** until we cross the finish point of the objects's geometry.
- * This negative value, together with [RoadObjectGeometry.length]
- * can be used to calculate the distance since the start of an object.
+ * @param distanceToStart remaining distance to the start of the object.
```

Comment:
>**this value will be negative** until we cross the finish point of the objects's geometry.
 * This negative value, together with [RoadObjectGeometry.length]
 * can be used to calculate the distance since the start of an object.

Does this comment still apply? If yes, could you reintroduce it?

Reply:
cc @etl @SiarheiFedartsou

Reply:
Looks like the comment [isn't actual](https://github.com/mapbox/mapbox-navigation-native/blob/master/src/eh/manager.cpp#L326)

Reply:
It will be still negative for route alerts, but it will be zero for eHorizon objects(but we are going to fix it some future and have behavior similar to what we have with route alerts).